src/http/notification/NotificationSubscriptionHttpHandler.ts

```diff
@@ -0,0 +1,127 @@
+import type { EventEmitter } from 'events';
+import type { CredentialsExtractor } from '../../authentication/CredentialsExtractor';
+import type { PermissionReader } from '../../authorization/PermissionReader';
+import { AccessMode } from '../../authorization/permissions/Permissions';
+import type { PermissionMap } from '../../authorization/permissions/Permissions';
+import type { OperationHandlerInput } from '../../http/ldp/OperationHandler';
+import { OkResponseDescription } from '../../http/output/response/OkResponseDescription';
+import type { ResponseDescription } from '../../http/output/response/ResponseDescription';
+import { RepresentationMetadata } from '../../http/representation/RepresentationMetadata';
+import { getLoggerFor } from '../../logging/LogUtil';
+import type { Subscription } from '../../notification/Subscription';
+import type { SubscriptionHandler } from '../../notification/SubscriptionHandler';
+import type { Topic } from '../../notification/Topic';
+import { OperationHttpHandler } from '../../server/OperationHttpHandler';
+import type { OperationHttpHandlerInput } from '../../server/OperationHttpHandler';
+import type { KeyValueStorage } from '../../storage/keyvalue/KeyValueStorage';
+import { BadRequestHttpError } from '../../util/errors/BadRequestHttpError';
+import { NotImplementedHttpError } from '../../util/errors/NotImplementedHttpError';
+import { IdentifierSetMultiMap } from '../../util/map/IdentifierMap';
+import { AS } from '../../util/Vocabularies';
+import type { ResourceIdentifier } from '../representation/ResourceIdentifier';
+
+/**
+ * Handles the negotiation of notification channels
+ */
+export class NotificationSubscriptionHttpHandler extends OperationHttpHandler {
+  protected readonly logger = getLoggerFor(this);
+
+  private readonly subscriptionHandlers: Map<string, SubscriptionHandler<Subscription>> = new Map();
+
+  public constructor(
+    private readonly credentialsExtractor: CredentialsExtractor,
+    private readonly permissionReader: PermissionReader,
+    private readonly notificationStorage: KeyValueStorage<string, Topic>,
+    private readonly source: EventEmitter,
+    subscriptionHandlers: SubscriptionHandler<Subscription>[],
+  ) {
+    super();
+    for (const handler of subscriptionHandlers) {
+      this.subscriptionHandlers.set(handler.getType(), handler);
+    }
+    this.source.on(AS.Create, async(id: ResourceIdentifier): Promise<void> => this.onResourceChanged(id, AS.Create));
+    this.source.on(AS.Update, async(id: ResourceIdentifier): Promise<void> => this.onResourceChanged(id, AS.Update));
+    this.source.on(AS.Delete, async(id: ResourceIdentifier): Promise<void> => this.onResourceChanged(id, AS.Delete));
```

Comment:
These trigger whenever _any_ document is updated, including internal documents. Internal documents should be ignored.